core/src/main/scala/org/apache/spark/sql/execution/columnar/impl/ColumnFormatEntry.scala

```diff
@@ -113,68 +112,50 @@ final class ColumnFormatKey(private[columnar] var partitionId: Int,
 
   def getColumnIndex: Int = columnIndex
 
-  override def hashCode(): Int = Murmur3_x86_32.hashLong(
-    (columnIndex.toLong << 32L) | (uuid.hashCode & 0xffffffffL), partitionId)
+  override def hashCode(): Int = Murmur3_x86_32.hashInt(
```

Comment:
Earlier code was dealing with three integers: string hash, columnIndex and partitionId. For that it was ORing two to create a long then combining with third using mumur's hashLong.

Now it is dealing with a long and two integers. So now combining the long and int first. Used the ClientResolverUtils method instead of murmur for that which has been added for SnappyData and has been tested to work best (clash and performance combination) and used in ObjectHashMapAccessor. Finally combined the two integers with murmur.

Ideally it could use mixK1, mixH1 and fmix methods from murmur but those are private in Spark's implementation (see Murmur3_x86_32.hashUnsafeBytes) but I don't think clashes will be an issue here so an even simpler combination is likely to also work as well.

Reply:
ok. Thanks.